src/backend/renderer/gles2/mod.rs

```diff
@@ -487,16 +490,10 @@ impl Gles2Renderer {
             if gl_version < version::GLES_3_0 && !exts.iter().any(|ext| ext == "GL_EXT_unpack_subimage") {
                 return Err(Gles2Error::GLExtensionNotSupported(&["GL_EXT_unpack_subimage"]));
             }
-            // required for instanced damage rendering
-            if gl_version < version::GLES_3_0
-                && !(exts.iter().any(|ext| ext == "GL_EXT_instanced_arrays")
-                    && exts.iter().any(|ext| ext == "GL_EXT_draw_instanced"))
-            {
-                return Err(Gles2Error::GLExtensionNotSupported(&[
-                    "GL_EXT_instanced_arrays",
-                    "GL_EXT_draw_instanced",
-                ]));
-            }
+            // Check if GPU supports instanced rendering.
+            let supports_instancing = gl_version >= version::GLES_3_0
+                || (exts.iter().any(|ext| ext == "GL_EXT_instanced_arrays")
+                    && exts.iter().any(|ext| ext == "GL_EXT_draw_instanced"));
```

Comment:
While @Drakulix has recommended on IRC to add an environment variable to override this, I've chosen against doing so because I don't see much practical application for it.

When specifically testing instanced/uninstanced one is likely already compiling from source anyway and in that scenario just setting it to a fixed value is trivial. Even while writing this patch I didn't see any need to make this logic more complicated.

Reply:
The practical application is:
1. downstream depending on a released version of smithay
2. encounters a bug
3. wants to verify if it happens with both renderer paths

Sure you could also add a "patch"-statement in your Cargo.toml for this case.

But I agree, that both code paths are similar enough (especially given they both use the same shader), that this kind of scenario should be very rare. So this is likely fine the way it is.

Reply:
Yeah that's basically the only thing I could come up with too, but if you're on older hardware opting into instanced simply isn't an option. And if you're on newer hardware then it really wouldn't be of much help when the garbage path of non-instanced rendering works, especially because the "backup" plan is much more likely to be broken than what basically everyone uses.